hugo/assets/ts/widgets/menu.ts

```diff
@@ -0,0 +1,92 @@
+import { BaseWidget } from './base-widget';
+
+export class Menu extends BaseWidget {
+    public static readonly NAME = 'menu';
+
+    dropdowns: NodeListOf<HTMLElement>;
+
+    constructor(element: HTMLElement) {
+        super(element);
+
+        this.dropdowns = this.element.querySelectorAll<HTMLElement>('[data-menu-dropdown]');
+    }
+
+    public static registerWidget() {
+        if (window.app !== undefined) {
+            window.app.addWidget({
+                name: Menu.NAME,
+                load: Menu.attachWidgetToElements,
+            });
+        }
+    }
+
+    public static attachWidgetToElements(container: HTMLElement | Document) {
+        const elements = container.querySelectorAll<HTMLElement>(`[data-${ Menu.NAME }]`);
+        elements.forEach((element) => {
+            const newWidget = new Menu(element);
+            newWidget.init();
+        });
+    }
+
+    public init() {
+        this.dropdowns.forEach((dropdownElement: HTMLElement) => {
+            const trigger = dropdownElement;
+            const target = this.getTarget(trigger);
+
+            if (target) {
+                trigger.addEventListener('click', (e) => {
+                    e.preventDefault();
+
+                    if (trigger.classList.contains('is-active')) {
+                        this.close(trigger, target);
+                    } else {
+                        this.open(trigger, target);
```

Comment:
maybe also close other dropdowns inside the menu when one is opening? in case we have more in the future